src/main/java/com/google/roomies/database/Database.java

```diff
@@ -0,0 +1,76 @@
+package com.google.roomies;
+
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map; 
+
+interface Database {
+  /**
+  * Gets the database instance.
+  *
+  * @throws IOException if database cannot be initialized
+  */
+  public Firestore getDatabase() throws IOException;
+
+  /**
+  * Sets the database instance.
+  * Used to set database to a mock for testing.
+  *
+  * @throws IOException if database cannot be initialized
+  */
+  public void setDatabaseForTest(Firestore database);
```

Comment:
Moved to databaseFactory. Keeping the setDatabaseForTest for time sake, since we have that mock working right now. Will keep in mind for future implementations